### PR TITLE
Implement support for Aspire

### PR DESCRIPTION
**Pull request by tmat**
Follow ups:
 - [ ] https://github.com/dotnet/aspire/pull/6090
 - [x] https://github.com/dotnet/sdk/issues/43946

### CODE REVIEW

**Comment by tmat** [comment]
@phil-allen-msft @davidfowl Ready!

File: src/BuiltInTools/dotnet-watch/Aspire/AspireServiceFactory.cs [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+            if (projectLaunchInfo.Arguments != null)
+            {
+                arguments.AddRange(projectLaunchInfo.Arguments);
+            }
+
+            return new()
+            {
+                IsRootProject = false,
+                ProjectPath = projectLaunchInfo.ProjectPath,
+                WorkingDirectory = projectLauncher.EnvironmentOptions.WorkingDirectory, // TODO: Should DCP protocol specify?
```
Comment (phil-allen-msft) [inline src/BuiltInTools/dotnet-watch/Aspire/AspireServiceFactory.cs:108]: How would we get an answer on these 'TODO' comments?

File: src/BuiltInTools/dotnet-watch/Program.cs [hunk 2]
```diff
@@ -159,8 +159,15 @@ internal Watcher CreateWatcher(IRuntimeProcessLauncherFactory? runtimeProcessLau
             var projectGraph = TryReadProject(rootProjectOptions, reporter);
             if (projectGraph != null)
             {
+                var rootProject = projectGraph.GraphRoots.Single();
+
                 // use normalized MSBuild path so that we can index into the ProjectGraph
-                rootProjectOptions = rootProjectOptions with { ProjectPath = projectGraph.GraphRoots.Single().ProjectInstance.FullPath };
+                rootProjectOptions = rootProjectOptions with { ProjectPath = rootProject.ProjectInstance.FullPath };
+
+                if (rootProject.GetCapabilities().Contains("Aspire"))
```
Comment (phil-allen-msft) [inline src/BuiltInTools/dotnet-watch/Program.cs:167]: Do you want to use `AppHostProjectCapability` from above?

File: src/BuiltInTools/dotnet-watch/Aspire/AspireServiceFactory.cs (hunk 1 above)
Comment (tmat) [reply src/BuiltInTools/dotnet-watch/Aspire/AspireServiceFactory.cs:108]: @davidfowl

**Comment by tmat** [comment]
Moving to 9.0.2xx: https://github.com/dotnet/sdk/pull/43978

**Pull request closed without merging**